pilopt/tests/optimizer.rs

```diff
@@ -567,12 +634,159 @@ fn inline_chain_of_substitutions() {
     col witness a;
     col witness b;
     col x = N::a + N::y;
-    col y = N::x + N::b;
+    col witness y;
+    N::y = N::x + N::b;
     col m = N::x - N::y;
     N::a * N::b = 10;
     N::m * N::a = 1;
 "#;
 
-    let optimized = optimize(analyze_string::<GoldilocksField>(input).unwrap()).to_string();
+    let optimized = optimize(analyze_string::<GoldilocksField>(input).unwrap(), 3).to_string();
+    assert_eq!(optimized, expectation);
+}
+
+#[test]
+fn witness_column_degree_limitation() {
+    let input = r#"namespace N(65536);
+    col witness a;
+    
+    // This column can be inlined (degree 2)
+    col witness x;
+    x = a * a;
+    
+    // This column CANNOT be inlined directly because
+    // its definition already has degree 4 > MAX_DEGREE (3)
+    col witness high_degree;
+    high_degree = a * a * a * a;
```

Comment:
This is an odd case, running the optimizer with a degree bound lower than the current bound. I suspect we could panic in this case, because right now the output has a degree higher than the parameter passed to the optimizer.